awx/main/tasks.py

```diff
@@ -1917,6 +1928,60 @@ def build_env(self, inventory_update, **kwargs):
         env['AWX_PRIVATE_DATA_DIR'] = kwargs.get('private_data_dir', '')
         return env
 
+    def get_inventory_plugin_name(self, inventory_update):
+        if inventory_update.source in Inventory.SOURCE_PLUGINS:
+            return Inventory.SOURCE_PLUGINS[inventory_update.source]
+        if inventory_update.source == 'custom':
+            # TODO: allow user to specify which plugin to use
+            return None  # intentionally allow other defaults, like ini
+        return 'script'
+
+    def build_inventory(self, inventory_update, **kwargs):
+        src = inventory_update.source
+        if src in Inventory.SOURCE_PLUGINS and Version(kwargs['ansible_version']) >= Version('2.4.0'):
+            cred_data = kwargs.get('private_data_files', {}).get('credentials', '')
+            cloud_credential = cred_data.get(inventory_update.credential, '')
+            # Build inventory contents specific to each plugin type
+            if src == 'openstack':
```

Comment:
This isn't a requirement for this PR, but just something on my mind as we continue to work on this script -> migration in general.

I've always really disliked the way that our inventory script building is a giant chain of `if src == 'xyz'` checks in `tasks.py` here.  It would be _really_ nice to take this opportunity to move this config building logic into separate, testable code elsewhere, maybe something like `InventoryUpdate.build_dynamic_inventory_file(**kwargs)`, which would write a file and return a path for `-i` ?